### PR TITLE
Update Rails and RSpec configs and generated files

### PR DESCRIPTION
**Pull request by andrewgarner**
This is proposed as an alternative to #607.

It brings the configs and files generated by `$ rails app:update` and `$ rails generate rspec:install` in line with the versions of Rails and RSpec we depend on.

This means that:
- we adopt the community driven conventions and opinions provided by the frameworks
- we minimise the diff when we repeat the process when upgrading to future versions of the frameworks

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -3,26 +3,32 @@ require 'pathname'
...
   puts '== Installing dependencies =='
   system! 'gem install bundler --conservative'
   system('bundle check') || system!('bundle install')
-  system! 'bin/yarn'
+
+  puts '== Installing JavaScript dependencies =='
+  system('bin/yarn')
 
   puts "\n== Setting environment =="
   unless File.exist?('.env')
```
Comment (benthorner) [inline bin/setup]: This should be unnecessary now.

**Comment by benthorner** [comment]
Nooooo...pointless comments :-( But yes, it's a logical approach.

File: bin/setup (hunk 1 above)
Comment (andrewgarner) [reply bin/setup]: Indeed, good spot!

**Comment by benthorner** [comment]
In fact, we can remove .env, I think?

**Comment by andrewgarner** [comment]
`.env` isn't tracked by git so removing the step from `bin/setup` should be good!

**Comment by benthorner** [comment]
But also the gem :-)

**Comment by benthorner** [comment]
dotenv-rails

**Pull request merged**